src/EditorFeatures/Core.Wpf/InlineRename/InlineRenameService.cs

```diff
@@ -84,6 +92,38 @@ internal class InlineRenameService : IInlineRenameService
                 _asyncListener);
 
             return new InlineRenameSessionInfo(ActiveSession);
+
+            static InlineRenameSessionInfo IsReadOnlyOrCannotNavigateToSpan(IInlineRenameInfo renameInfo, Document document, CancellationToken cancellationToken)
+            {
+                if (renameInfo is IInlineRenameInfoWithFileRename renameInfoWithFileRename)
+                {
+                    var workspace = document.Project.Solution.Workspace;
+                    var navigationService = workspace.Services.GetService<IDocumentNavigationService>();
```

Comment:
may want to #nullable enable this file and then this should become GetRequiredService

feel free to leave out though if the rest of the file is really painful to annotate